hail/src/main/scala/is/hail/annotations/RegionValueBuilder.scala

```diff
@@ -175,6 +192,11 @@ class RegionValueBuilder(var region: Region) {
     advance()
   }
 
+  def setArrayIndex(newI: Int) {
+    assert(typestk.top.isInstanceOf[PArray])
+    indexstk(0) = newI
```

Comment:
Should this be `indexstk.push(newI)`?

Reply:
No, the index stack refers the position of the cursor at the 'current level' of the stack.